Assets/MixedRealityToolkit/Providers/BaseInputDeviceManager.cs

```diff
@@ -61,36 +155,97 @@ public abstract class BaseInputDeviceManager : BaseDataProvider<IMixedRealityInp
         /// <param name="useSpecificType">Only register pointers with a specific type.</param>
         protected virtual IMixedRealityPointer[] RequestPointers(SupportedControllerType controllerType, Handedness controllingHand)
         {
-            var pointers = new List<IMixedRealityPointer>();
+            var returnPointers = new List<IMixedRealityPointer>();
 
-            if ((Service != null) &&
-                (InputSystemProfile != null) &&
-                InputSystemProfile.PointerProfile != null)
+            for (int i = 0; i < pointerConfigurations.Length; i++)
             {
-                for (int i = 0; i < InputSystemProfile.PointerProfile.PointerOptions.Length; i++)
-                {
-                    var pointerProfile = InputSystemProfile.PointerProfile.PointerOptions[i];
+                var option = pointerConfigurations[i].profile;
 
-                    if (((pointerProfile.ControllerType & controllerType) != 0) &&
-                        (pointerProfile.Handedness == Handedness.Any || pointerProfile.Handedness == Handedness.Both || pointerProfile.Handedness == controllingHand))
+                if (option.ControllerType.HasFlag(controllerType) && option.Handedness.HasFlag(controllingHand))
+                {
+                    var pointerCache = pointerConfigurations[i].cache;
+                    if (pointerCache.Count > 0)
                     {
-                        var pointerObject = Object.Instantiate(pointerProfile.PointerPrefab);
-                        MixedRealityPlayspace.AddChild(pointerObject.transform);
-                        var pointer = pointerObject.GetComponent<IMixedRealityPointer>();
-
-                        if (pointer != null)
+                        var p = pointerCache.Pop();
+                        if (p != null && p is MonoBehaviour pointerGameObject)
```

Comment:
>if (p != null && p is MonoBehaviour pointerGameObject) [](start = 24, length = 54)

This is probably not the type of feedback being requested, but I did wanna leave a little comment about this one (i.e. it feels weird that we'd store stuff in the cache that we wouldn't consider later - I guess this is primarily to defend against cases where the game object is destroyed out from under us through other mechanisms right?)

Reply:
@wiwei no happy for the feedback. I just didn't want nit stuff atm like forgot to add comment etc. 

So I agree, but I'm not sure how to overcome this. The pointer is a GameObject and anyone with a reference to it could at some pointer potentially destroy it. So it shouldn't ever be null but we have to check against it.